### DISCUSSION TITLE
Data limitation with Micro App

### DISCUSSION THREAD

**Discussion by mdkberry** [deta/cloud-docs#384] [Help] [answered]
I want to use Deta for FastAPI with stored/updatable csv files that contain a lot of OHCL chart data for stock, pull the data on request and serve it via FastAPI. Some csv files for 1h data can be around 10mb or more. 

I saw that "HTTP Payload size limit is 5.5 MB" is a limitation in the Micro App. does this also apply if I store the csv data using Drive when it is being read and written to (using pandas and dataframes), not moved? (I havent figured out how Micro Deta and Drive Deta connect yet exactly, I was just looking at feasibility of this working for me)

**Comment by AyushSehrawat** [comment, marked as answer]
HTTP Payload size limit of 5.5mb will apply when you try to show it on the webpage... If you are like just internally getting the csv file, it would be fine, but if you will be returning the csv larger than 5.5mb it will show error...

**Reply by AyushSehrawat** [reply]
returning means like serving the file via fastapi or uploading a csv file via fastapi framework. Like 
```py
    from fastapi.responses import StreamingResponse
    import io
    
    @app.get("/get_csv")
    async def get_csv():
    
       df = pandas.DataFrame(dict(col1 = 1, col2 = 2))
    
       stream = io.StringIO()
    
       df.to_csv(stream, index = False)
    
       response = StreamingResponse(iter([stream.getvalue()]),
                            media_type="text/csv"
       )
    
       response.headers["Content-Disposition"] = "attachment; filename=export.csv"

       return response
``` 
Something like this with a file of 5.5 mb or more will give payload error, 

Same goes if you are trying to upload a csv file via FastAPI ( like with `UploadFile` )

**Reply by mdkberry** [reply]
thanks. that would impact my requirements.
